src/steps/resource-manager/advisor/executionHandlers/recommendations.ts

```diff
@@ -0,0 +1,134 @@
+  executionContext: IntegrationStepContext,
+): Promise<void> {
+  const { instance, logger, jobState } = executionContext;
+  const accountEntity = await getAccountEntity(jobState);
+  const webLinker = createAzureWebLinker(accountEntity.defaultDomain as string);
+  const client = new AdvisorClient(instance.config, logger);
+
+  await client.iterateRecommendations(async (recommendation) => {
+    await jobState.addEntity(
+      createRecommendationEntity(webLinker, recommendation),
+    );
+  });
+}
+
+export async function buildAssesmentToRecommendationRelationship(
+  executionContext: IntegrationStepContext,
+): Promise<void> {
+  const { jobState } = executionContext;
+
+  await jobState.iterateEntities(
+    { _type: AdvisorEntities.RECOMMENDATION._type },
+    async (recommendationEntity) => {
+      const recommendation = getRawData<ResourceRecommendationBase>(
+        recommendationEntity,
+      )!;
+      if (recommendation.resourceMetadata?.source) {
+        const assessmentEntity = await jobState.findEntity(
+          recommendation.resourceMetadata.source,
+        );
+        if (assessmentEntity) {
+          await jobState.addRelationship(
+            createDirectRelationship({
+              _class: AdvisorRelationships.ASSESSMENT_IDENTIFIED_FINDING._class,
+              from: assessmentEntity,
+              to: recommendationEntity,
+              properties: {
+                _type: AdvisorRelationships.ASSESSMENT_IDENTIFIED_FINDING._type,
+              },
+            }),
+          );
+        }
+      }
+    },
+  );
+}
+
+export async function buildResourceToRecommendationRelationship(
+  executionContext: IntegrationStepContext,
+): Promise<void> {
+  const { jobState } = executionContext;
+
+  await jobState.iterateEntities(
+    { _type: AdvisorEntities.RECOMMENDATION._type },
+    async (recommendationEntity) => {
+      const recommendation = getRawData<ResourceRecommendationBase>(
+        recommendationEntity,
+      )!;
+      if (recommendation.resourceMetadata?.resourceId) {
+        const resourceEntity = await jobState.findEntity(
+          recommendation.resourceMetadata.resourceId,
+        );
+        if (resourceEntity) {
+          await jobState.addRelationship(
+            createDirectRelationship({
+              _class: AdvisorRelationships.ANY_RESOURCE_HAS_FINDING._class,
+              from: resourceEntity,
+              to: recommendationEntity,
+              properties: {
+                _type: AdvisorRelationships.ANY_RESOURCE_HAS_FINDING._type,
+              },
+            }),
+          );
+        }
+      }
+    },
+  );
+}
+
+export const recommendationSteps: AzureIntegrationStep[] = [
+  {
+    id: AdvisorSteps.RECOMMENDATIONS,
+    name: 'Recommendations',
+    entities: [AdvisorEntities.RECOMMENDATION],
+    relationships: [],
+    dependsOn: [
+      STEP_AD_ACCOUNT,
+      SecuritySteps.ASSESSMENTS,
+      ...getResourceManagerSteps().executeFirstSteps,
+    ],
```

Comment:
This should only depend on the `STEP_AD_ACCOUNT` step:

```suggestion
    dependsOn: [
      STEP_AD_ACCOUNT,
    ],
```